src/CompilingMatcher.php

```diff
@@ -61,6 +61,6 @@ public static function match(ConstraintInterface $constraint, $operator, $versio
             $function = self::$compiledCheckerCache[$cacheKey];
         }
 
-        return $function($version, $version[0] === 'd' && 'dev-' === substr($version, 0, 4));
+        return $function($version, $version[0] === 'd' && strpos($version, 'dev-') === 0);
```

Comment:
is it an improvement? do you have some numbers?

I doubt that `strpos` (search in the entire haystack) is better than `substr` (cut the first 4 chars).

Reply:
The point is mainly better code maintainability. A performance difference is not noteworthy but in case of a substring you have to create a new string which costs more than iterating a whole short string too.
It shouldn't be used on big strings, thats' correct though.

Reply:
Some more context in phpinspectionsea plugin for PHPStorm I use
https://github.com/kalessil/phpinspectionsea/blob/master/docs/control-flow.md#substr-could-be-replaced-with-strpos